app/models/ability.rb

```diff
@@ -4,27 +4,24 @@ class Ability
   def initialize(user)
     if user
       can :manage, User, id: user.id
-      can :manage, GroupedIssue do |group|
-        user.is_owner_of?(group.website)
-      end
-      can :manage, Issue do |issue|
-        user.is_owner_of?(issue.group.website)
-      end
-      can :manage, Message do |message|
-        user.is_owner_of?(message.issue.group.website)
-      end
+      can :manage, GroupedIssue
+      can :manage, Issue
+      can :manage, Message
```

Comment:
In this case here you gave access to any user to access any grouped_issue, issue or message. It doesn't matter if the user is a part of the website that these records belong to. You should add some specs for permissions and for these check if the user is a member of the website these records belong to.